eclair-core/src/main/scala/fr/acinq/eclair/router/Sync.scala

```diff
@@ -86,59 +92,67 @@ object Sync {
     implicit val sender: ActorRef = ctx.self // necessary to preserve origin when sending messages to other actors
     ctx.sender ! TransportHandler.ReadAck(r)
 
-    Metrics.ReplyChannelRange.Blocks.withTag(Tags.Direction, Tags.Directions.Incoming).record(r.numberOfBlocks)
-    Metrics.ReplyChannelRange.ShortChannelIds.withTag(Tags.Direction, Tags.Directions.Incoming).record(r.shortChannelIds.array.size)
+    d.sync.get(origin.nodeId) match {
+      case None =>
+        log.info("received unsolicited reply_channel_range with {} channels", r.shortChannelIds.array.size)
+        d // we didn't request a sync from this node, ignore
+      case Some(currentSync) if currentSync.remainingQueries.isEmpty && r.shortChannelIds.array.isEmpty =>
+        log.info("received empty reply_channel_range, sync is complete")
+        d.copy(sync = d.sync - origin.nodeId)
```

Comment:
This isn't how the spec indicates that we should detect the last `reply_channel_range` message:

> the final reply_channel_range message:
>    MUST have first_blocknum plus number_of_blocks equal or greater than the query_channel_range first_blocknum plus number_of_blocks.

Reply:
Right, this is probably a leftover made to handle nodes that didn't correctly signal termination?
I'll look into it.
Note that we'll be bringing back the `complete` field which will greatly simplify our lives.

Reply:
This case doesn't handle the normal termination of `reply_channel_range`, it deals with a special case where the *first* `reply_channel_range` is empty (meaning the remote node doesn't want to send us anything basically). Otherwise we'll have something in `remainingQueries` and we'll wait for these to complete before we consider the sync done. I had to add this because we have a test case for it. 

An interesting note is that we don't detect the end of a `reply_channel_range` at all: we rely on `reply_short_channel_ids_end` and the fact that our `remainingQueries` is empty. _That's different from the spec and should be fixed_. In practice it will very likely work (we've always had this behavior and we've been fine) and it's orthogonal to this PR since it was our previous behavior. I think that can be fixed in a later PR (and we can wait to rely on the `complete` field for a cleaner signal that the sync is complete once we correctly set it again).

Reply:
Maybe you should include your reply as a `TODO` note here?

Reply:
Done in https://github.com/ACINQ/eclair/pull/1587/commits/8f292b9f7dc2bb40e65cc576f05c04a6722ce5ab and added to my todo list.